template_reports/office_renderer/context_extractor.py

```diff
@@ -51,11 +52,20 @@ def extract_context_keys(template) -> dict[str, list[str]]:
 
     simple_fields = set()
     object_fields = set()
+    loop_variables = set()  # Store loop iterator variables to ignore them
 
     # Build a list of all texts on all slides.
     texts = []
     for slide in prs.slides:
         for shape in slide.shapes:
+            # Check for loop directives
+            if hasattr(shape, "text_frame"):
+                loop_var, loop_collection = extract_loop_directive(shape.text_frame.text)
+                if loop_var and loop_collection:
+                    # Add collection to object_fields
```

Comment:
don't add to the object_fields here, as that is done afterwards at the bottom (line 91-94). only add loop var to the loop variables